lib/galaxy/datatypes/msa.py

```diff
@@ -11,6 +11,56 @@
 log = logging.getLogger(__name__)
 
 
+class InfernalCM(Text):
+    file_ext = "cm"
+
+    MetadataElement(name="number_of_models", default=0, desc="Number of covariance models",
+                    readonly=True, visible=True, optional=True, no_value=0)
+
+    MetadataElement(name="cm_version", default="1/a", desc="Infernal Covariance Model version",
+                    readonly=True, visible=True, optional=True, no_value=0)
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = get_file_peek(dataset.file_name, is_multi_byte=is_multi_byte)
+            if dataset.metadata.number_of_models == 1:
+                dataset.blurb = "1 model"
+            else:
+                dataset.blurb = "%s models" % dataset.metadata.number_of_models
+            dataset.peek = get_file_peek(dataset.file_name, is_multi_byte=is_multi_byte)
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disc'
+
+    def sniff(self, filename):
+        """
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname( 'infernal_model.cm' )
+        >>> InfernalCM().sniff( fname )
+        True
+        >>> fname = get_test_fname( 'test.mz5' )
+        >>> InfernalCM().sniff( fname )
+        False
+        """
+        with open(filename, 'r') as f:
+            first_line = f.readline()
+
+        if first_line.startswith("INFERNAL"):
```

Comment:
You could do `return first_line.startswith("INFERNAL")`

Reply:
Done. Imo it takes the same time for a computer to run but twice the time for a human to understand ;-)